frontend/src/components/PendingMatchingPage.js

```diff
@@ -1,20 +1,74 @@
 import {
     Box,
-    Typography
+    Typography, 
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    Button,
+    DialogTitle,
 } from "@mui/material";
-import { useLocation } from "react-router-dom";
+import { useLocation, Navigate, Link } from "react-router-dom";
+import { io } from "socket.io-client";
+import {useState, useEffect} from "react";
 
 function PendingMatchPage() {
     const location = useLocation()
-    const { diff } = location.state
+    const { diff, name } = location.state
+
+    const [roomId, setRoomId] = useState("")
+    const [foundMatch, setFoundMatch] = useState(false)
+    const [matchName, setMatchName] = useState("")
+    const [noMatchFound, setNoMatchFound] = useState(false);
+
+    useEffect(() => {
+        const socket = io("http://localhost:8001");
+
+        socket.on("connect", () => {
+            console.log('socket is connected on client side', socket.id, 'with this id');
+
+            socket.emit('match', name, diff);
+
+            const timeout = setTimeout(() => {
+                socket.emit('delete', diff);
+                setNoMatchFound(true);
+            }, 30000);
+
+            socket.on('room', (match_name, val) => {
+                console.log('printing room id inside', val);
+                console.log('user name is', name);
+                console.log('match name is', match_name);
+                clearTimeout(timeout);
+                setFoundMatch(true);
+                setMatchName(match_name);
+                var url = '/room/' + val;
+                setRoomId(url);
+            });
+    
+            socket.on('delete confirm', (val) => {
+                console.log(val);
```

Comment:
Logs should be more descriptive, on reading the val how will we know what its about?